lib/Resizable/index.tsx

```diff
@@ -0,0 +1,123 @@
+import type { PropsWithChildren } from "react";
+import React, { useCallback, useEffect, useRef } from "react";
+import { keepValueWithinRange, toPixels } from "../helpers";
+
+export interface ResizableProps {
+  defaultWidth?: number | string;
+  minWidth?: number | string;
+  maxWidth?: number | string;
+  useGutterOffset?: boolean;
+  containerWidth?: number | string;
+}
+
+/**
+ * A wrapper for making a given child element resizable
+ * @param children The element to be resized
+ * @param defaultWidth The width the child should initially render to. If min and max widths are provided, then the provided width will be massaged to meet those constraints
+ * @param maxWidth The maximum width the element can be resized to
+ * @param minWidth The minimum width the element can be resized to
+ * @param useGutterOffset Toggle whether we should take the width of the gutter into account for calculating the width of the resized item. This can be handy depending on the CSS being applied to or around the child element
+ * @param rest Contains params that need converting such as the containerWidth
+ */
+export function Resizable({
+  children,
+  defaultWidth = "50%",
+  maxWidth,
+  minWidth,
+  useGutterOffset = false,
+  ...rest
+}: PropsWithChildren<ResizableProps>) {
+  const containerWidth = toPixels(
+    rest.containerWidth ?? document.body.offsetWidth
```

Comment:
Why is `containerWidth` stored in `rest` but other properties aren't? 🤔

Reply:
Yeah this doesn't look great. I was lazily looking for a way to dynamically set the default value for `containerWidth` if it was undefined but I didn't want to do it as a default param in case a falsy value such as `0` gets passed in and then overridden by my default.

I'll clean this up!